modules/pubProvidedSystem.js

```diff
@@ -0,0 +1,52 @@
+/**
+ * This module adds Publisher Provided ids support to the User ID module
+ * The {@link module:modules/userId} module is required.
+ * @module modules/pubProvidedSystem
+ * @requires module:modules/userId
+ */
+
+import {submodule} from '../src/hook.js';
+import * as utils from "../src/utils";
+
+const MODULE_NAME = 'pubProvided';
+
+/** @type {Submodule} */
+export const pubProvidedSubmodule = {
+  /**
+   * used to link submodule with config
+   * @type {string}
+   */
+  name: MODULE_NAME,
+
+  /**
+   * decode the stored id value for passing to bid request
+   * @function
+   * @param {string} value
+   * @returns {{pubProvided: array}} or undefined if value doesn't exists
+   */
+  decode(value) {
+    const res = value ? {pubProvided: value} : undefined;
+    utils.logInfo('PubProvided: Decoded value ' + JSON.stringify(res));
+    return res;
+  },
+
+  /**
+   * performs action to obtain id and return a value.
+   * @function
+   * @param {SubmoduleParams} [configParams]
+   * @returns {{id: array}}
+   */
+  getId(configParams) {
+    let res = [];
```

Comment:
i don't fully understand the use case that a publisher would have multiple sources of their own ID. the source should be their own domain, so while they may have multiple IDs (`eids[].uids[]`) why would they need more than one element in the `eids` array?

Reply:
@jdwieland8282 @bretg

Reply:
The pub might be working with a 3rd party, who has done some probabilistic, or deterministic identification on their behalf, and they have a second uuid based off their own tech stack. That determination is publisher driven.  The pub could be sending two ids for monetization on different platforms Device, desktop, CTV.

Reply:
Use case is that a publisher has both:
- a site-specific ID they generated on their own
- and a DMP vendor who doesn't have a Prebid.js User ID Sub-Module